full-service/src/db/wallet_db.rs

```diff
@@ -29,6 +29,8 @@ impl diesel::r2d2::CustomizeConnection<SqliteConnection, diesel::r2d2::Error>
             }
             if self.enable_foreign_keys {
                 conn.batch_execute("PRAGMA foreign_keys = ON;")?;
+            } else {
+                conn.batch_execute("PRAGMA foreign_keys = OFF;")?;
```

Comment:
The default bundled sqlite3 version has foreign keys enabled by default. I think it is significant tech debt that the tests don't run with this turned on and I suggest fixing it gets prioritized.